packages/playground/website/public/plugin-proxy.php

```diff
@@ -338,17 +338,56 @@ function ($curl, $body) use (&$extra_headers_sent, $default_response_headers) {
         $branch = strtolower( $_GET['wordpress-branch'] );
```

Comment:
While this is undocumented, should it be renamed to `wordpress-artifact` or similar given it now allows for more than just branches? Naming things suggestions are welcome.

Reply:
I'd love to see this all documented, ideally before the previous PR merged.
`git` talks about `refs` and `shas`, where a branch and tag are both a kind of `ref`.

perhaps the query arg could be something like `&git-ref=wordpress/wordpress-develop#0ac256d5fb` or `&git-ref=wordpress/gutenberg#trunk`

we could allow only specific repositories, but this syntax would be flexible enough to allow more than initially planned without changing the public interface

Reply:
There's already `repo` and `name` but it assumes a plugin release rather than a WordPress install.

https://github.com/WordPress/wordpress-playground/blob/eb0985e0a0e301bd6617f7bcec523a61a5b4270c/packages/playground/website/public/plugin-proxy.php#L330-L336

I could use `WordPress/WordPress=artifact` although it's potentially confusing as parameter names are case sensitive.

Reply:
if we have `repo` and `name` then what about `ref` as the third one?

Reply:
> if we have `repo` and `name` then what about `ref` as the third one?

We could do that as there's already some magic around the two

https://github.com/WordPress/wordpress-playground/blob/eb0985e0a0e301bd6617f7bcec523a61a5b4270c/packages/playground/website/public/plugin-proxy.php#L279-L280

So check for `isset($_GET['org']) && isset($_GET['repo']) && isset($_GET['ref'])`?

Although I'd like to understand the purpose better, it's unlikely we'll accept other WordPress forks as `wordpress.zip`. https://github.com/Woofpress/Woofpress will need to set up their own playground, for example.

Reply:
Why wouldn't that work for Woofpress/Woofpress, though?

Related – eventually I want to try hosting a generic [CORS proxy](https://github.com/WordPress/wordpress-playground/issues/1467) and forks may be able to use that. That is, unless GitHub rate limits these requests. But even then, it may still work via an [in-browser git clone](https://adamadam.blog/2024/06/21/cloning-a-git-repository-from-a-web-browser-using-fetch/).

Reply:
@peterwilsoncc this is unimportant and tangential to the discussion, but passing multiple arguments to `isset()` is effectively the same as calling it in an `&&` chain.

```php
} else if ( isset( $_GET['org'], $_GET['repo'], $_GET['ref'], $_GET['workflow'], $_GET['pr'], $_GET['artifact'] ) )
```

https://www.php.net/manual/en/function.isset.php

> So check for isset($_GET['org']) && isset($_GET['repo']) && isset($_GET['ref'])?

We can probably default `$_GET['ref']` so it's not necessary. Specifying a repo should return the latest `HEAD` from its main branch, no?

Would the `$_GET['ref']` and `$_GET['pr']` arguments be mutually exclusive?